plugins/list/plugin.js

```diff
@@ -760,6 +761,14 @@
 		if ( last && last.type == CKEDITOR.NODE_ELEMENT && last.is( 'br' ) )
 			last.remove();
 
+		// remove all empty inlines in the fragment
+
+		while ( emptyInline = frag.getLast( function ( node ) {
```

Comment:
This is a little cheeky. Any reason you chose to do the assignment as the conditional in the while loop?